src/json.hpp

```diff
@@ -8800,64 +8800,149 @@ class basic_json
-        long double str_to_float_t(long double* /* type */, char** endptr) const
+        long double strtojnum(const char *str) const
         {
-            return std::strtold(reinterpret_cast<typename string_t::const_pointer>(m_start), endptr);
-        }
+            long double result = 0;
+            char cp = *str;
+            int exp = 0; // exponent
+            {
+                const bool negative_sign = cp == '-';
 
-        /*!
-        @brief parse floating point number
+                if (cp == '-' or cp == '+')
+                {
+                    ++str;
+                }
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+                // read in fractional part of number, until an 'e' is reached.
+                // count digits after decimal point.
+                while (nl_isdigit(cp = *str))
+                {
+                     result = result * 10 + (cp - '0');
+                     ++str;
+                }
 
-        @param[in] type  the @ref number_float_t in use
+                if (cp == '.')
+                {
+                     while (nl_isdigit(cp = *++str))
+                     {
+                         result = result * 10 + (cp - '0');
+                         --exp;
+                     }
+                }
 
-        @param[in,out] endptr  recieves a pointer to the first character after
-        the number
+                // if negative number, reverse sign
+                if (negative_sign)
+                {
+                    result *= -1;
+                }
+            }
 
-        @return the floating point number
-        */
-        double str_to_float_t(double* /* type */, char** endptr) const
-        {
-            return std::strtod(reinterpret_cast<typename string_t::const_pointer>(m_start), endptr);
-        }
+            // read in explicit exponent and calculate real exponent.
+            if (*str == 'e' or *str == 'E')
+            {
+                cp = *++str;
 
-        /*!
-        @brief parse floating point number
+                const bool negative_exp = cp == '-'; // read in exponent sign (+/-)
+                const bool plus_or_minus = (cp == '-' or cp == '+');
+                if (plus_or_minus)
+                {
+                     cp = *++str;
+                }
 
-        This function (and its overloads) serves to select the most approprate
-        standard floating point number parsing function based on the type
-        supplied via the first parameter.  Set this to @a
-        static_cast<number_float_t*>(nullptr).
+                int count = 0; // exponent calculation
+                if (not nl_isdigit(cp))
+                {
+                     if (plus_or_minus)
+                     {
+                         *--str;
```

Comment:
Thanks for taking a look! @nlohmann want to incorp these into your feature branch?